src/main/java/org/zalando/nakadi/controller/SubscriptionController.java

```diff
@@ -75,16 +78,40 @@ public SubscriptionController(final SubscriptionDbRepository subscriptionReposit
         } catch (final DuplicatedSubscriptionException e) {
             try {
                 return new ResponseEntity<>(getExistingSubscription(subscriptionBase), HttpStatus.OK);
-            } catch (final NakadiException ex) {
+            } catch (final ServiceUnavailableException ex) {
+                LOG.error("Error occurred during fetching existing subscription", ex);
+                return create(e.asProblem(), request);
+            } catch (final NoSuchSubscriptionException | InternalNakadiException ex) {
                 LOG.error("Error occurred during fetching existing subscription", ex);
                 return create(INTERNAL_SERVER_ERROR, ex.getProblemMessage(), request);
             }
-        } catch (final InternalNakadiException e) {
+        } catch (final NakadiException e) {
             LOG.error("Error occurred during subscription creation", e);
             return create(e.asProblem(), request);
         }
     }
 
+    @RequestMapping(method = RequestMethod.GET)
+    public ResponseEntity<?> listSubscriptions(
+            @Nullable @RequestParam(value = "owning_application", required = false) final String owningApplication,
+            final NativeWebRequest request) {
+
+        if (!featureToggleService.isFeatureEnabled(HIGH_LEVEL_API)) {
+            return new ResponseEntity<>(NOT_IMPLEMENTED);
+        }
+
+        try {
+            final List<Subscription> subscriptions = owningApplication == null ?
+                    subscriptionRepository.listSubscriptions() :
+                    subscriptionRepository.listSubscriptionsForOwningApplication(owningApplication);
+            return status(OK).body(subscriptions);
```

Comment:
Regarding wrapper object - true. Will fix it.
Regarding owning_application - it was requested during API review. I also think it makes sense.
https://github.com/zalando/nakadi/pull/353/files#diff-4492c6b26ee0a37518728ec1585420efR579